src/java.base/share/classes/java/time/LocalDateTime.java

```diff
@@ -1177,16 +1177,16 @@ public LocalDateTime plus(TemporalAmount amountToAdd) {
     @Override
     public LocalDateTime plus(long amountToAdd, TemporalUnit unit) {
         if (unit instanceof ChronoUnit chronoUnit) {
-            switch (chronoUnit) {
-                case NANOS: return plusNanos(amountToAdd);
-                case MICROS: return plusDays(amountToAdd / MICROS_PER_DAY).plusNanos((amountToAdd % MICROS_PER_DAY) * 1000);
-                case MILLIS: return plusDays(amountToAdd / MILLIS_PER_DAY).plusNanos((amountToAdd % MILLIS_PER_DAY) * 1000_000);
-                case SECONDS: return plusSeconds(amountToAdd);
-                case MINUTES: return plusMinutes(amountToAdd);
-                case HOURS: return plusHours(amountToAdd);
-                case HALF_DAYS: return plusDays(amountToAdd / 256).plusHours((amountToAdd % 256) * 12);  // no overflow (256 is multiple of 2)
-            }
-            return with(date.plus(amountToAdd, unit), time);
+            return switch (chronoUnit) {
+                case NANOS     -> plusNanos(amountToAdd);
+                case MICROS    -> plusDays(amountToAdd / MICROS_PER_DAY).plusNanos((amountToAdd % MICROS_PER_DAY) * 1000);
+                case MILLIS    -> plusDays(amountToAdd / MILLIS_PER_DAY).plusNanos((amountToAdd % MILLIS_PER_DAY) * 1000_000);
+                case SECONDS   -> plusSeconds(amountToAdd);
+                case MINUTES   -> plusMinutes(amountToAdd);
+                case HOURS     -> plusHours(amountToAdd);
+                case HALF_DAYS -> plusDays(amountToAdd / 256).plusHours((amountToAdd % 256) * 12);
+                default -> with(date.plus(amountToAdd, unit), time); // no overflow (256 is multiple of 2)
```

Comment:
The comment is misplaced.

Reply:
Hi Naoto. Thanks for spotting this. I've corrected that now. Please see commit 2ae4a57